src/backend/base/langflow/components/embeddings/amazon_bedrock.py

```diff
@@ -20,28 +19,96 @@ class AmazonBedrockEmbeddingsComponent(LCModelComponent):
+            "Only needed for temporary credentials. "
+            "Usually set in Python code as the environment variable 'AWS_SESSION_TOKEN'.",
+        ),
+        SecretStrInput(
             name="credentials_profile_name",
             display_name="Credentials Profile Name",
             advanced=True,
+            info="The name of the profile to use from your "
+            "~/.aws/credentials file. "
+            "If not provided, the default profile will be used.",
+        ),
+        DropdownInput(
+            name="region_name",
+            display_name="Region Name",
+            value="us-east-1",
+            options=[
+                "us-west-2",
+                "us-west-1",
+                "us-gov-west-1",
+                "us-gov-east-1",
+                "us-east-2",
+                "us-east-1",
+                "sa-east-1",
+                "me-south-1",
+                "me-central-1",
+                "il-central-1",
+                "eu-west-3",
+                "eu-west-2",
+                "eu-west-1",
+                "eu-south-2",
+                "eu-south-1",
+                "eu-north-1",
+                "eu-central-2",
+                "eu-central-1",
+                "cn-northwest-1",
+                "cn-north-1",
+                "ca-west-1",
+                "ca-central-1",
+                "ap-southeast-5",
+                "ap-southeast-4",
+                "ap-southeast-3",
+                "ap-southeast-2",
+                "ap-southeast-1",
+                "ap-south-2",
+                "ap-south-1",
+                "ap-northeast-3",
+                "ap-northeast-2",
+                "ap-northeast-1",
+                "ap-east-1",
+                "af-south-1",
+            ],
+            info="The AWS region where your Bedrock resources are located.",
+        ),
+        MessageTextInput(
+            name="endpoint_url",
+            display_name="Endpoint URL",
+            advanced=True,
+            info="The URL of the AWS Bedrock endpoint to use.",
         ),
-        MessageTextInput(name="region_name", display_name="Region Name", value="us-east-1"),
-        MessageTextInput(name="endpoint_url", display_name=" Endpoint URL", advanced=True),
     ]
 
     outputs = [
         Output(display_name="Embeddings", name="embeddings", method="build_embeddings"),
     ]
 
     def build_embeddings(self) -> Embeddings:
-        if self.aws_access_key:
+        if self.aws_access_key_id or self.aws_secret_access_key:
             import boto3
 
             session = boto3.Session(
-                aws_access_key_id=self.aws_access_key,
-                aws_secret_access_key=self.aws_secret_key,
+                aws_access_key_id=self.aws_access_key_id,
+                aws_secret_access_key=self.aws_secret_access_key,
+                aws_session_token=self.aws_session_token,
             )
         elif self.credentials_profile_name:
             import boto3
```

Comment:
Maybe a try/except for the boto3 import before the conditions to keep only one import at the top